src/repo.rs

```diff
@@ -0,0 +1,618 @@
+#![allow(dead_code)]
+
+#![feature(proc_macro)]
+#![feature(alloc)]
+
+// Conract doesn't need standard library and the `main` function.
+// `cargo test` requires `std` and it provided the `main` which is why the "std" feature should be turned on for `cargo test`
+#![cfg_attr(not(feature="std"), no_main)]
+#![cfg_attr(not(feature="std"), no_std)]
+
+extern crate alloc;
+extern crate pwasm_std;
+extern crate pwasm_abi;
+extern crate pwasm_abi_derive;
+
+use pwasm_abi::eth::EndpointInterface;
+use pwasm_std::{storage, Vec};
+use pwasm_std::hash::{H256};
+
+struct Entry {
+    key: H256,
+    value: [u8; 32]
+}
+
+struct Storage {
+    table: Vec<Entry>,
+}
+
+/// TODO: optimise by impl hashtable
+impl Storage {
+    fn with_capacity(cap: usize) -> Storage {
```

Comment:
why do you need public with_capacity here?
isn't capacity obvious from the usage?

Reply:
you mean better to just hardcode it?

Reply:
yeah, why bothering doing second `with_capacity` function calling the first one despite you use only one value for an argument